core/src/main/scala/kafka/server/KafkaHealthcheck.scala

```diff
@@ -92,7 +92,6 @@ class KafkaHealthcheck(private val brokerId: Int,
 
     override def handleSessionEstablishmentError(error: Throwable): Unit = {
       fatal("Could not establish session with zookeeper", error)
-      System.exit(-1)
```

Comment:
KafkaController has the following comment:
"handleSessionEstablishmentError in KafkaHealthCheck should System.exit and log the error."

If we no longer System.exit, we should probably modify this.

Also, if we don't exit with "-1" code, what's the new behavior when I start a broker and it fails to connect to ZK?